cirq/google/engine/engine_timeslot.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import dataclasses
+
+from typing import Optional
+from cirq.google.engine.client.quantum_v1alpha1.gapic import enums
+
+_DEFAULT_TYPE = enums.QuantumProcessor.TimeSlotType.TIME_SLOT_TYPE_UNSPECIFIED
+
+
+@dataclasses.dataclass(frozen=True)
+class EngineTimeSlot:
+    """A python wrapping of a Quantum Engine timeslot.
+  """
+    start_seconds: int
```

Comment:
WDYT of datetime for start and end times? Is there a reason to prefer seconds?

Reply:
Well, this would require some additional conventions.  The proto defines this as a Timestamp which is seconds since epoch:

https://cs.corp.google.com/piper///depot/google3/google/protobuf/timestamp.proto?l=125

If we define this in datetime, we have to figure out how to translate this to datetime with timezone information and watch out that we don't screw up when serializing / deserializing.  (For instance, if one researcher serializes to JSON and another in the project but in a different time zone gets the timeslot and uses it.)

It's totally doable, but it adds some complexity here.  Are you sure we need this information?

Reply:
We don't need it, but I'm concerned about the usability if this is the interface for viewing a processor's schedule. Time slots are read-only, and IIUC `datetime.fromtimestamp()` will convert to local time, so it sounds like we're just pushing that onto the users. Not a big deal yet though and I can wait to see where this goes.

Reply:
Ok, changed to datetime.  I figure its easier to change now than later, and we can probably create a from_proto() function once the protos are in cirq.